processing/src/test/java/io/druid/segment/incremental/IncrementalIndexStorageAdapterTest.java

```diff
@@ -458,4 +458,157 @@ public Object apply(Cursor cursor)
         new ArrayList<>()
     );
   }
+
+  @Test
+  public void testCursoringAndSnapshot() throws Exception
+  {
+    final IncrementalIndex index = indexCreator.createIndex();
+    final long timestamp = System.currentTimeMillis();
+
+    for (int i = 0; i < 2; i++) {
+      index.add(
+          new MapBasedInputRow(
+              timestamp,
+              Lists.newArrayList("billy"),
+              ImmutableMap.<String, Object>of("billy", "v0" + i)
+          )
+      );
+    }
+
+    final StorageAdapter sa = new IncrementalIndexStorageAdapter(index);
+
+    Sequence<Cursor> cursors = sa.makeCursors(
+        null, new Interval(timestamp - 60_000, timestamp + 60_000), VirtualColumns.EMPTY, Granularities.ALL, false
+    );
+
+    Sequences.toList(
```

Comment:
Please include something here to make sure that the code actually gets executed. It's possible that `cursors` is empty and none of these assertions are run.

Either checking the return value of Sequences.toList, or incrementing an AtomicInteger inside the map fn and checking it after Sequences.toList returns, should be enough.